showcase-app/plugin/identityPlugin.js

```diff
@@ -44,69 +85,72 @@ export default class IdentityWidget {
 
     if (message.type === 'active-identity-response') {
       if (message.data) {
-        let idpOrWebID = message.data.idp;
-
-        if (message.data.webID) {
-          const idps = await this._getIDPsFromWebID(message.data.webID)
-
-          if (idps.length) {
-            idpOrWebID = idps[0];
-          }
-        }
-
-        this.activeIdentity = {
-          ...message.data,
-          idpOrWebID,
-        };
+        // Make the received identity actual
+        this.activeIdentity = message.data;
 
+        // Fire the change handler
+        // The Solid App can subscribe to this event to handle any changes in the app to trigger required behaviour
         this.identityChangedHandler({
           ...this.activeIdentity,
         });
       }
     }
-  }
-
-  async _getIDPsFromWebID(webId) {
-    const myEngine = new QueryEngine();
-    const bindingsStream = await myEngine.queryBindings(`
-    SELECT ?idp WHERE {
-      <${webId}> <http://www.w3.org/ns/solid/terms#oidcIssuer> ?idp
-    } LIMIT 10`, {
-      sources: [webId],
-    });
-
-    const bindings = await bindingsStream.toArray();
-    return bindings.map(a => a.get('idp').value);
-  }
+  };
 
   /**
    * PUBLIC METHODS (API)
    */
 
+  /** @typedef {{ webID: string }} WebID*/
+  /** @typedef {{ idp: string }} IDP*/
+  /** @typedef {( WebID | IDP )} DataType*/
+  /** @typedef {{ data: DataType }} IDP*/
+  /**
+   * Gets all identities currently available in the extension
+   * @returns {Promise<DataType[]>} - the identities from the extension
+   */
   getIdentities = () => {
+    // This method promisifies feedback from the async flow imposed by usage of messaging ports
     return new Promise((resolve) => {
+      // handle incoming messages from the Chrome extension
       const handleRequest = (message) => {
         if (message.type === 'all-identities-response') {
+          // We resolve the promise with the required data and then perform cleanup
           resolve(message.data);
           this.port.onMessage.removeListener(handleRequest);
         }
       };
 
+      // Subscribe new message handler and then fire the message to the extension to kickstart the start of the entire cycle, requesting all identities
       this.port.onMessage.addListener(handleRequest);
       this.port.postMessage({ type: 'request-identities' });
     });
-  }
+  };
 
+  /**
+   * @callback identityChangedCallback
+   * @param {Identity} identity - the new identity that is active
+   */
+  /**
+   * Call this method with the function you want to execute whenever a profile change occurs
+   * @param {identityChangedCallback} callback - the callback invoked upon identity change
+   */
```

Comment:
How come there are two description after each other?

Reply:
the second one is for the function on the next line.
the first one is the callback typing of the parameter used in the function